beacon_node/lighthouse_network/src/discovery/enr.rs

```diff
@@ -159,14 +159,22 @@ pub fn build_or_load_enr<E: EthSpec>(
     local_key: Keypair,
     config: &NetworkConfig,
     enr_fork_id: &EnrForkId,
+    custody_group_count: Option<u64>,
     next_fork_digest: [u8; 4],
     spec: &ChainSpec,
 ) -> Result<Enr, String> {
     // Build the local ENR.
     // Note: Discovery should update the ENR record's IP to the external IP as seen by the
     // majority of our peers, if the CLI doesn't expressly forbid it.
     let enr_key = CombinedKey::from_libp2p(local_key)?;
-    let mut local_enr = build_enr::<E>(&enr_key, config, enr_fork_id, next_fork_digest, spec)?;
+    let mut local_enr = build_enr::<E>(
+        &enr_key,
+        config,
+        enr_fork_id,
+        custody_group_count,
+        next_fork_digest,
+        spec,
+    )?;
```

Comment:
ah i didnt think this was needed because i thought we'd use the one from disk

https://github.com/sigp/lighthouse/blob/b43e0b446cd1d8a3f5b3c4a9f614f950ef672aa8/beacon_node/lighthouse_network/src/discovery/enr.rs#L123-L128

But then the above scenario evaluates to false because we set the CGC to the default value when building the ENR, so it doesn't survive across restarts.

Your fix makes sense 👍